plugins/ai/provider/gis_ai/classes/privacy/provider.php

```
@@ -0,0 +1,103 @@
+            'status' => 'privacy:metadata:aiprovider_gis_ai_logs:status',
+            'contextid' => 'privacy:metadata:aiprovider_gis_ai_logs:contextid',
+            'courseid' => 'privacy:metadata:aiprovider_gis_ai_logs:courseid',
+        ], 'privacy:metadata:aiprovider_gis_ai_logs');
+        return $collection;
+    }
+
+    /**
+     * Get contexts for a given user id.
+     */
+    public static function get_contexts_for_userid(int $userid): contextlist {
+        global $DB;
```

PHPMD (2.15.0)

Avoid unused local variables such as '$DB'. (undefined)

(UnusedLocalVariable)